test/cri-containerd/clone_test.go

```diff
@@ -0,0 +1,404 @@
+
+	if exitCode != 0 || len(errorMsg) != 0 {
+		t.Fatalf("Failed execution inside container %s with error: %s, exitCode: %d", containerID, errorMsg, exitCode)
+	} else {
+		t.Logf("Exec(container: %s) stdout: %s, stderr: %s, exitCode: %d\n", containerID, output, errorMsg, exitCode)
+	}
+}
+
+// A simple test to just create a template container and then create one
+// cloned container from that template.
+func Test_CloneContainer_WCOW(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	client := newTestRuntimeClient(t)
+
+	pullRequiredImages(t, []string{imageWindowsNanoserver})
+
+	templatePodID, templateContainerID := createTemplateContainer(ctx, t, client, getTemplatePodConfig("templatepod"), getTemplateContainerConfig("templatecontainer"))
+	defer removePodSandbox(t, client, ctx, templatePodID)
+	defer stopPodSandbox(t, client, ctx, templatePodID)
+	defer removeContainer(t, client, ctx, templateContainerID)
+	defer stopContainer(t, client, ctx, templateContainerID)
+
+	clonedPodID, clonedContainerID := createClonedContainer(ctx, t, client, templatePodID, templateContainerID, 1)
+	defer removePodSandbox(t, client, ctx, clonedPodID)
+	defer stopPodSandbox(t, client, ctx, clonedPodID)
+	defer removeContainer(t, client, ctx, clonedContainerID)
+	defer stopContainer(t, client, ctx, clonedContainerID)
+
+	verifyContainerExec(ctx, t, client, clonedContainerID)
+}
+
+// A test for creating multiple clones(5 clones) from one template container.
+func Test_MultiplClonedContainers_WCOW(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	client := newTestRuntimeClient(t)
+	nClones := 3
+
+	pullRequiredImages(t, []string{imageWindowsNanoserver})
+
+	// create template pod & container
+	templatePodID, templateContainerID := createTemplateContainer(ctx, t, client, getTemplatePodConfig("templatepod"), getTemplateContainerConfig("templatecontainer"))
+	defer removePodSandbox(t, client, ctx, templatePodID)
+	defer stopPodSandbox(t, client, ctx, templatePodID)
+	defer removeContainer(t, client, ctx, templateContainerID)
+	defer stopContainer(t, client, ctx, templateContainerID)
+
+	// create multiple clones
+	clonedContainers := []string{}
+	for i := 0; i < nClones; i++ {
+		clonedPodID, clonedContainerID := createClonedContainer(ctx, t, client, templatePodID, templateContainerID, i)
+		// cleanup
+		defer removePodSandbox(t, client, ctx, clonedPodID)
+		defer stopPodSandbox(t, client, ctx, clonedPodID)
+		defer removeContainer(t, client, ctx, clonedContainerID)
+		defer stopContainer(t, client, ctx, clonedContainerID)
+		clonedContainers = append(clonedContainers, clonedContainerID)
+	}
+
+	for i := 0; i < nClones; i++ {
+		verifyContainerExec(ctx, t, client, clonedContainers[i])
+	}
+}
+
+// Test if a normal container can be created inside a clond pod alongside the cloned
+// container.
+// TODO(ambarve): This doesn't work as of now. Enable this test when the bug is fixed.
```

Comment:
Is there still a bug here?

Reply:
Yes, I did some investigations on this but didn't get enough time to fix this yet. We can either keep this test as disabled and merge or we can remove the test. What do you think?

Reply:
I think we should remove the test then and possible add a block in the code to prevent this case if possible?

Reply:
Is the bug on our end or OS related? Do you think we'll remember to put it back in later on? We merged the GMSA PR with hyper-v tests even though that wasn't possible at the time. I just had the first line in the test skip it, that made it so it was just a one line change to enable it when the scenario lit up. Either way works but it would save rewriting this test again when the bug is fixed.

Reply:
I'm fine with disabling it too and having a comment that says that it's disabled due to a bug

Reply:
sounds good

Reply:
So the current comment looks good? I will keep the name of the function as it is so that the test is disabled.